File: VB/Q210958/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Drawing.Drawing2D
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo

Namespace Q210958
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
			Me.categoriesTableAdapter.Fill(Me.nwindDataSet.Categories)

		End Sub

		Private Sub gridView1_CustomDrawCell(ByVal sender As Object, ByVal e As RowCellCustomDrawEventArgs) Handles gridView1.CustomDrawCell
			Dim view As GridView = CType(sender, GridView)
			Dim viewInfo As GridViewInfo = CType(view.GetViewInfo(), GridViewInfo)
			Dim ci As GridColumnsInfo = viewInfo.ColumnsInfo
			If (Not view.IsDataRow(e.RowHandle)) OrElse e.RowHandle = view.FocusedRowHandle OrElse ci.FirstColumnInfo.Column IsNot e.Column Then
				Return
			End If
			Dim even As Boolean = e.RowHandle Mod 2 = 0
			Dim hStyle As HatchStyle = If(even, HatchStyle.BackwardDiagonal, HatchStyle.Cross)
			Dim fColor As Color = If(even, Color.AliceBlue, Color.AntiqueWhite)
			Dim bColor As Color = If(even, Color.Aqua, Color.Aquamarine)
			Dim row As New Rectangle(viewInfo.ViewRects.Rows.Left, e.Bounds.Y, viewInfo.ViewRects.Rows.Width, e.Bounds.Height)
			Using brush As Brush = New HatchBrush(hStyle, fColor, bColor)
				e.Cache.FillRectangle(brush, row)
			End Using
		End Sub
	End Class
End Namespace

File: CS/Q210958/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using System.Drawing.Drawing2D;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;

namespace Q210958 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Categories' table. You can move, or remove it, as needed.
            this.categoriesTableAdapter.Fill(this.nwindDataSet.Categories);

        }

        private void gridView1_CustomDrawCell(object sender, RowCellCustomDrawEventArgs e) {
            GridView view = (GridView)sender;
            GridViewInfo viewInfo = (GridViewInfo)view.GetViewInfo();
            GridColumnsInfo ci = viewInfo.ColumnsInfo;
            if (!view.IsDataRow(e.RowHandle) || e.RowHandle == view.FocusedRowHandle ||
                ci.FirstColumnInfo.Column != e.Column) 
                return;
            bool even = e.RowHandle % 2 == 0;
            HatchStyle hStyle = even ? HatchStyle.BackwardDiagonal : HatchStyle.Cross;
            Color fColor = even ? Color.AliceBlue : Color.AntiqueWhite;
            Color bColor = even ? Color.Aqua : Color.Aquamarine;
            Rectangle row = new Rectangle(viewInfo.ViewRects.Rows.Left, e.Bounds.Y,
                viewInfo.ViewRects.Rows.Width, e.Bounds.Height);
            using (Brush brush = new HatchBrush(hStyle, fColor, bColor)) {
                e.Graphics.FillRectangle(brush, row);
            }
        }
    }
}